### PR TITLE
factor out Dependency creation in Lib so it can be reused

### PR DESCRIPTION
**Pull request by ajrnz**
the diff looks a bit messy but it's most just cut and paste

### CODE REVIEW

File: scalalib/src/mill/scalalib/Lib.scala [hunk 1]
```diff
@@ -27,47 +26,51 @@ object Lib{
...
-    val flattened = deps.map{
+  def depToDependency(dep: Dep, scalaVersion: String, platformSuffix: String = ""): Dependency =
+    dep match {
       case Dep.Java(dep, cross) =>
         dep.copy(
           module = dep.module.copy(
             name =
               dep.module.name +
-              (if (!cross) "" else platformSuffix)
+                (if (!cross) "" else platformSuffix)
```
Comment (lihaoyi) [inline scalalib/src/mill/scalalib/Lib.scala]: Could you format these the old way?

Comment (ajrnz) [reply scalalib/src/mill/scalalib/Lib.scala]: Sure, missed that. That's IntelliJ. Do you have a set of IntelliJ defaults for this or just have auto formatting turned off?

**Comment by ajrnz** [comment]
BTW this should be fixed now. Let me know if not

File: scalalib/src/mill/scalalib/Lib.scala [hunk 2]
```diff
@@ -54,20 +41,36 @@ object Lib{
           module = dep.module.copy(
             name =
               dep.module.name +
-              (if (!cross) "" else platformSuffix) +
-              "_" + scalaBinaryVersion(scalaVersion)
+                (if (!cross) "" else platformSuffix) +
+                "_" + scalaBinaryVersion(scalaVersion)
```
Comment (lihaoyi) [inline scalalib/src/mill/scalalib/Lib.scala]: @ajrnz looks like you're still making unnecessary formatting changes here 😛I don't know what your IDE is doing, but just make sure these are gone by the time your code ends up on a PR. Edit your code in the github web editor if you have to

**Comment by lihaoyi** [comment]
Tests looked flaky, re-triggering
